--- a/dbms/tests/queries/0_stateless/00113_group_array.sql
+++ b/dbms/tests/queries/0_stateless/00113_group_array.sql
@@ -14,10 +14,6 @@ SELECT count(), sum(toUInt64(ns)), max(t
 SELECT count(), sum(toUInt64(ns[1])), max(toUInt64(ns[1])), sum(toUInt64(ns[2]))/10 FROM (SELECT intDiv(number, 100) AS k, groupArray([toString(number), toString(number*10)]) AS ns FROM test.numbers_mt GROUP BY k) ARRAY JOIN ns;
 SELECT count(), sum(ns[1]), max(ns[1]), sum(ns[2])/10 FROM (SELECT intDiv(number, 100) AS k, groupArray([number, number*10]) AS ns FROM test.numbers_mt GROUP BY k) ARRAY JOIN ns;
 
-SELECT count(), sum(ns), max(ns) FROM (SELECT intDiv(number, 100) AS k, groupArray(number) AS ns FROM remote('127.0.0.{1,2}', 'test', 'numbers_mt') GROUP BY k) ARRAY JOIN ns;
-SELECT count(), sum(toUInt64(ns)), max(toUInt64(ns)) FROM (SELECT intDiv(number, 100) AS k, groupArray(toString(number)) AS ns FROM remote('127.0.0.{1,2}', 'test', 'numbers_mt') GROUP BY k) ARRAY JOIN ns;
-SELECT count(), sum(toUInt64(ns[1])), max(toUInt64(ns[1])), sum(toUInt64(ns[2]))/10 FROM (SELECT intDiv(number, 100) AS k, groupArray([toString(number), toString(number*10)]) AS ns FROM remote('127.0.0.{1,2}', 'test', 'numbers_mt') GROUP BY k) ARRAY JOIN ns;
-
 DROP TABLE test.numbers_mt;
 CREATE TABLE test.numbers_mt (number UInt64) ENGINE = Log;
 INSERT INTO test.numbers_mt SELECT * FROM system.numbers LIMIT 1, 1048575;
@@ -27,11 +23,6 @@ SELECT roundToExp2(number) AS k, length(
 SELECT roundToExp2(number) AS k, length(groupArray(1)(hex(number) AS i)), length(groupArray(1024)(i)), length(groupArray(65536)(i)) AS s FROM test.numbers_mt GROUP BY k ORDER BY k LIMIT 9, 11;
 SELECT roundToExp2(number) AS k, length(groupArray(1)([hex(number)] AS i)), length(groupArray(1024)(i)), length(groupArray(65536)(i)) AS s FROM test.numbers_mt GROUP BY k ORDER BY k LIMIT 9, 11;
 
-SELECT '';
-SELECT roundToExp2(number) AS k, length(groupArray(1)(number AS i)), length(groupArray(1500)(i)), length(groupArray(70000)(i)) AS s FROM remote('127.0.0.{1,2}', 'test', 'numbers_mt') GROUP BY k ORDER BY k LIMIT 9, 11;
-SELECT roundToExp2(number) AS k, length(groupArray(1)(hex(number) AS i)), length(groupArray(1500)(i)), length(groupArray(70000)(i)) AS s FROM remote('127.0.0.{1,2}', 'test', 'numbers_mt') GROUP BY k ORDER BY k LIMIT 9, 11;
-SELECT roundToExp2(number) AS k, length(groupArray(1)([hex(number)] AS i)), length(groupArray(1500)(i)), length(groupArray(70000)(i)) AS s FROM remote('127.0.0.{1,2}', 'test', 'numbers_mt') GROUP BY k ORDER BY k LIMIT 9, 11;
-
 DROP TABLE test.numbers_mt;
 
 -- Check binary compability:
--- a/dbms/tests/queries/0_stateless/00113_group_array.reference
+++ b/dbms/tests/queries/0_stateless/00113_group_array.reference
@@ -17,9 +17,6 @@
 1000000	500000500000	1000000
 1000000	500000500000	1000000	500000500000
 1000000	500000500000	1000000	500000500000
-2000000	1000001000000	1000000
-2000000	1000001000000	1000000
-2000000	1000001000000	1000000	1000001000000
 
 512	1	512	512
 1024	1	1024	1024
@@ -54,37 +51,3 @@
 131072	1	1024	65536
 262144	1	1024	65536
 524288	1	1024	65536
-
-512	1	1024	1024
-1024	1	1500	2048
-2048	1	1500	4096
-4096	1	1500	8192
-8192	1	1500	16384
-16384	1	1500	32768
-32768	1	1500	65536
-65536	1	1500	70000
-131072	1	1500	70000
-262144	1	1500	70000
-524288	1	1500	70000
-512	1	1024	1024
-1024	1	1500	2048
-2048	1	1500	4096
-4096	1	1500	8192
-8192	1	1500	16384
-16384	1	1500	32768
-32768	1	1500	65536
-65536	1	1500	70000
-131072	1	1500	70000
-262144	1	1500	70000
-524288	1	1500	70000
-512	1	1024	1024
-1024	1	1500	2048
-2048	1	1500	4096
-4096	1	1500	8192
-8192	1	1500	16384
-16384	1	1500	32768
-32768	1	1500	65536
-65536	1	1500	70000
-131072	1	1500	70000
-262144	1	1500	70000
-524288	1	1500	70000
