Index: clickhouse/dbms/tests/queries/0_stateless/00506_union_distributed.reference
===================================================================
--- clickhouse.orig/dbms/tests/queries/0_stateless/00506_union_distributed.reference	2018-01-21 16:15:23.865615244 +0100
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,16 +0,0 @@
-3	8
-23	48
-33	68
-13	28
-3	8
-23	48
-33	68
-13	28
-3	8
-23	48
-33	68
-13	28
-3	8
-23	48
-33	68
-13	28
Index: clickhouse/dbms/tests/queries/0_stateless/00506_union_distributed.sql
===================================================================
--- clickhouse.orig/dbms/tests/queries/0_stateless/00506_union_distributed.sql	2018-01-21 16:15:23.865615244 +0100
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,24 +0,0 @@
-
--- https://github.com/yandex/ClickHouse/issues/1059
-
-DROP TABLE IF EXISTS test.union1;
-DROP TABLE IF EXISTS test.union2;
-DROP TABLE IF EXISTS test.union3;
-
-CREATE TABLE test.union1 ( date Date, a Int32, b Int32, c Int32, d Int32) ENGINE = MergeTree(date, (a, date), 8192);
-CREATE TABLE test.union2 ( date Date, a Int32, b Int32, c Int32, d Int32) ENGINE = Distributed(test_shard_localhost, 'test', 'union1');
-CREATE TABLE test.union3 ( date Date, a Int32, b Int32, c Int32, d Int32) ENGINE = Distributed(test_shard_localhost, 'test', 'union2');
-
-INSERT INTO test.union1 VALUES (1,  2, 3, 4, 5);
-INSERT INTO test.union1 VALUES (11,12,13,14,15);
-INSERT INTO test.union2 VALUES (21,22,23,24,25);
-INSERT INTO test.union3 VALUES (31,32,33,34,35);
-
-select b, sum(c) from ( select a, b, sum(c) as c from test.union2 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union2 where b>1 group by a, b ) as a group by b;
-select b, sum(c) from ( select a, b, sum(c) as c from test.union1 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union2 where b>1 group by a, b ) as a group by b;
-select b, sum(c) from ( select a, b, sum(c) as c from test.union1 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union1 where b>1 group by a, b ) as a group by b;
-select b, sum(c) from ( select a, b, sum(c) as c from test.union2 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union3 where b>1 group by a, b ) as a group by b;
-
-DROP TABLE test.union1;
-DROP TABLE test.union2;
-DROP TABLE test.union3;
Index: clickhouse/dbms/tests/queries/0_stateless/00506_union_distributed_zookeeper.reference
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ clickhouse/dbms/tests/queries/0_stateless/00506_union_distributed_zookeeper.reference	2018-01-21 16:15:23.861615237 +0100
@@ -0,0 +1,16 @@
+3	8
+23	48
+33	68
+13	28
+3	8
+23	48
+33	68
+13	28
+3	8
+23	48
+33	68
+13	28
+3	8
+23	48
+33	68
+13	28
Index: clickhouse/dbms/tests/queries/0_stateless/00506_union_distributed_zookeeper.sql
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ clickhouse/dbms/tests/queries/0_stateless/00506_union_distributed_zookeeper.sql	2018-01-21 16:15:23.861615237 +0100
@@ -0,0 +1,24 @@
+
+-- https://github.com/yandex/ClickHouse/issues/1059
+
+DROP TABLE IF EXISTS test.union1;
+DROP TABLE IF EXISTS test.union2;
+DROP TABLE IF EXISTS test.union3;
+
+CREATE TABLE test.union1 ( date Date, a Int32, b Int32, c Int32, d Int32) ENGINE = MergeTree(date, (a, date), 8192);
+CREATE TABLE test.union2 ( date Date, a Int32, b Int32, c Int32, d Int32) ENGINE = Distributed(test_shard_localhost, 'test', 'union1');
+CREATE TABLE test.union3 ( date Date, a Int32, b Int32, c Int32, d Int32) ENGINE = Distributed(test_shard_localhost, 'test', 'union2');
+
+INSERT INTO test.union1 VALUES (1,  2, 3, 4, 5);
+INSERT INTO test.union1 VALUES (11,12,13,14,15);
+INSERT INTO test.union2 VALUES (21,22,23,24,25);
+INSERT INTO test.union3 VALUES (31,32,33,34,35);
+
+select b, sum(c) from ( select a, b, sum(c) as c from test.union2 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union2 where b>1 group by a, b ) as a group by b;
+select b, sum(c) from ( select a, b, sum(c) as c from test.union1 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union2 where b>1 group by a, b ) as a group by b;
+select b, sum(c) from ( select a, b, sum(c) as c from test.union1 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union1 where b>1 group by a, b ) as a group by b;
+select b, sum(c) from ( select a, b, sum(c) as c from test.union2 where a>1 group by a,b UNION ALL select a, b, sum(c) as c from test.union3 where b>1 group by a, b ) as a group by b;
+
+DROP TABLE test.union1;
+DROP TABLE test.union2;
+DROP TABLE test.union3;
